test/src/module/command/expireTest.c

```diff
@@ -1252,6 +1252,540 @@ testRun(void)
+        TEST_RESULT_STR(
+            strLstJoin(strLstSort(storageListP(
+                storageTest, strNewFmt("%s/%s/%s", strPtr(archiveStanzaPath), "12-2", "0000000100000000")), sortOrderAsc), ", "),
+            archiveRemaining,
+            "no archives removed from latest except what was already removed");
+
+        //--------------------------------------------------------------------------------------------------------------------------
+        TEST_TITLE("error on expire last full backup in current db-id");
+
+        argList = strLstDup(argListAvoidWarn);
+        strLstAddZ(argList, "--set=20181119-152850F");
+        harnessCfgLoad(cfgCmdExpire, argList);
+
+        TEST_ERROR(
+            cmdExpire(), BackupSetInvalidError,
+            "full backup 20181119-152850F cannot be expired until another full backup has been created");
+
+        //--------------------------------------------------------------------------------------------------------------------------
+        TEST_TITLE("allow adhoc expire on last full backup in prior db-id");
+
+        argList = strLstDup(argListAvoidWarn);
+        strLstAddZ(argList, "--set=20181119-152800F");
+        harnessCfgLoad(cfgCmdExpire, argList);
+
+        TEST_RESULT_VOID(cmdExpire(), "adhoc expire last prior db-id backup");
+        TEST_RESULT_BOOL(
+            (storageExistsP(storageTest, strNewFmt("%s/20181119-152850F/" BACKUP_MANIFEST_FILE, strPtr(backupStanzaPath))) &&
+            !storageExistsP(
+                storageTest, strNewFmt("%s/20181119-152800F/" BACKUP_MANIFEST_FILE, strPtr(backupStanzaPath))) &&
+            !storageExistsP(
+                storageTest, strNewFmt("%s/20181119-152800F_20181119-152252D/" BACKUP_MANIFEST_FILE, strPtr(backupStanzaPath)))),
+            true, "only last prior backup removed");
+        harnessLogResult(
+            strPtr(strNewFmt(
+            "P00   INFO: expire adhoc backup set: 20181119-152800F, 20181119-152800F_20181119-152252D\n"
+            "P00   INFO: remove expired backup 20181119-152800F_20181119-152252D\n"
+            "P00   INFO: remove expired backup 20181119-152800F\n"
+            "P00   INFO: remove archive path: %s/repo/archive/db/9.4-1\n"
+            "P00 DETAIL: archive retention on backup 20181119-152850F, archiveId = 12-2, start = 000000010000000000000002\n"
+            "P00 DETAIL: no archive to remove, archiveId = 12-2", testPath())));
+
+        //--------------------------------------------------------------------------------------------------------------------------
+        TEST_TITLE("error on expire last full backup on disk");
+
+        argList = strLstDup(argListAvoidWarn);
+        strLstAddZ(argList, "--set=20181119-152850F");
+        harnessCfgLoad(cfgCmdExpire, argList);
+
+        TEST_ERROR(
+            cmdExpire(), BackupSetInvalidError,
+            "full backup 20181119-152850F cannot be expired until another full backup has been created");
+
+
```

Comment:
Remove extra linefeed.

Reply:
Fixed in later commit: 602acfc5e9780d1fea778a41ec9846afac90bf6d